frontend/services/allowance.ts

```diff
@@ -0,0 +1,96 @@
+import { Allowance, AllowancesTableColumns } from "@/@types/allowance";
+import { getFromLocalStorage, setInLocalStorage } from "../utils/localStorage";
+import { SortOrder } from "@/@types/common";
+import {
+  filterByAmount,
+  filterByAsset,
+  filterBySpender,
+  sortByExpiration,
+  sortBySubAccount,
+} from "@/utils/allowanceSorters";
+
+export const LOCAL_STORAGE_PREFIX = "allowances";
+
+type ListAllowances = (
+  tokenSymbol?: string,
+  column?: AllowancesTableColumns,
+  order?: SortOrder,
+) => Promise<Allowance[]>;
+
+export const listAllowances: ListAllowances = (tokenSymbol, column, order = SortOrder.ASC) => {
+  return new Promise((resolve) => {
+    const allowances = getFromLocalStorage<Allowance[]>(LOCAL_STORAGE_PREFIX);
+    if (!allowances || !Array.isArray(allowances)) resolve([]);
+
+    const filteredAllowances = tokenSymbol && allowances ? filterByAsset(tokenSymbol, allowances) : allowances;
+
+    if (column === AllowancesTableColumns.subAccount) {
+      resolve(sortBySubAccount(order, filteredAllowances || []));
+    }
+
+    if (column === AllowancesTableColumns.spender) {
+      resolve(filterBySpender(order, filteredAllowances || []));
+    }
+
+    if (column === AllowancesTableColumns.expiration) {
+      resolve(sortByExpiration(order, filteredAllowances || []));
+    }
+
+    if (column === AllowancesTableColumns.amount) {
+      resolve(filterByAmount(order, filteredAllowances || []));
+    }
+
+    resolve([]);
+  });
+};
+
+export function postAllowance(newAllowance: Allowance): Promise<Allowance> {
+  return new Promise((resolve) => {
+    let allowances = getFromLocalStorage<Allowance[]>(LOCAL_STORAGE_PREFIX);
+
+    if (!allowances || !Array.isArray(allowances)) {
+      allowances = [newAllowance];
+    }
+
+    if (Array.isArray(allowances)) {
+      allowances.push(newAllowance);
+    }
+
+    setTimeout(() => {
```

Comment:
why a time out? is easier to create a global function called sleep that will handle the wait time. and you can just add an await to that sleep function.